### PR TITLE
Chapter 10 Manage Users

### PR DESCRIPTION
**Pull request by Iskandar19**

### CODE REVIEW

**Comment by Iskandar19** [comment]
Ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,12 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
```
Comment (ghost) [inline app/controllers/users_controller.rb]: use except please

File: app/views/shared/_error_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      <div class="alert alert-danger">
+        <%= t "the_form_contains" %>
+        <%= pluralize @user.errors.count, "error" %>.
+      </div>
+      <ul>
+      <% @user.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+      <% end %>
+      </ul>
+    </div>
```
Comment (ghost) [inline app/views/shared/_error_messages.html.erb]: fix indent please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by Iskandar19** [comment]
ready

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, except: %i(new show create)
+  before_action :correct_user, only: [:edit, :update]
```
Comment (ghost) [inline app/controllers/users_controller.rb]:
```suggestion
  before_action :correct_user, only: %i(edit update)
```

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -20,10 +28,56 @@ def create
...
+  end
+
+  def update
+    @user = User.find_by(id: params[:id])
+    if @user.update(user_params)
+      flash[:success] = "Profile Updated"
+      redirect_to @user
+      # Handle a successful update.
+    else
+      render "edit"
```
Comment (ghost) [inline app/controllers/users_controller.rb]:
```suggestion
      render :edit
```

File: app/views/shared/_error_messages.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+        <div class="alert alert-danger">
+          <%= t "the_form_contains" %>
+          <%= pluralize @user.errors.count, "error" %>.
+        </div>
+        <ul>
+          <% @user.errors.full_messages.each do |msg| %>
+            <li><%= msg %></li>
+        <% end %>
+        </ul>
+    </div>
```
[the remark this answers is not in the document]
Comment (ghost) [reply app/views/shared/_error_messages.html.erb]: 2 space please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by Iskandar19** [comment]
ready

**Review by thaodtd-0665 — APPROVED** [review]

**Comment by thaodtd-0665** [comment]
merged

**Pull request merged**